src/scitoken.c

```diff
@@ -159,89 +181,116 @@ int ScitokenVerify(request_rec *r, const char *require_line, const void *parsed_
     return AUTHZ_DENIED;
   }
 
-  if(scitoken_deserialize(auth_line, &scitoken, null_ended_list, &err_msg)){
+  if(scitoken_deserialize(auth_line, &scitoken, (char const * const *) null_ended_list, &err_msg)){
     ap_log_rerror(APLOG_MARK, APLOG_INFO, 0, r, "Failed to deserialize scitoken");
     ap_log_rerror(APLOG_MARK, APLOG_INFO, 0, r, err_msg, r->uri);
     return AUTHZ_DENIED;
   }
   
-  char* issuer_ptr = NULL;
+  char *issuer_ptr = NULL;
+  char *sub_ptr = NULL;
+  char *aud_ptr = NULL;
+  char *scope_ptr = NULL;
+  char *wlcg_groups_ptr = NULL;
+  if(!scitoken_get_claim_string(scitoken, "sub", &sub_ptr, &err_msg)) {
+    apr_table_set(r->subprocess_env, "SCITOKEN_SUB", sub_ptr);
+  }
+  if(!scitoken_get_claim_string(scitoken, "aud", &aud_ptr, &err_msg)) {
+    apr_table_set(r->subprocess_env, "SCITOKEN_AUD", aud_ptr);
+  }
+  if(!scitoken_get_claim_string(scitoken, "scope", &scope_ptr, &err_msg)) {
+    apr_table_set(r->subprocess_env, "SCITOKEN_SCOPE", scope_ptr);
+  }
+  if(!scitoken_get_claim_string(scitoken, "wlcg.groups", &wlcg_groups_ptr, &err_msg)) {
+    apr_table_set(r->subprocess_env, "SCITOKEN_WLCG_GROUPS", wlcg_groups_ptr);
+  }
   if(scitoken_get_claim_string(scitoken, "iss", &issuer_ptr, &err_msg)) {
     ap_log_rerror(APLOG_MARK, APLOG_INFO, 0, r, "Failed to get issuer from token: %s\n",err_msg);
     ap_log_rerror(APLOG_MARK, APLOG_INFO, 0, r, err_msg, r->uri);
     return AUTHZ_DENIED;
   }
+  apr_table_set(r->subprocess_env, "SCITOKEN_ISS", issuer_ptr);
   
   //Preparing for enforcer test
   Enforcer enf;
   
   char hostname[1024];
-  const char* aud_list[2];
-  
-  // Get the hostname for the audience. It is using hostname(NOT domain name). Set payload accordingly
-  if (gethostname(hostname, 1024) != 0) {
-    ap_log_rerror(APLOG_MARK, APLOG_INFO, 0, r, "Failed to get hostname");
-    return AUTHZ_DENIED;
-  }
-  aud_list[0] = hostname;
-  aud_list[1] = NULL;
+  const char *aud_list[2];
 
-  if (!(enf = enforcer_create(issuer_ptr, aud_list, &err_msg))) {
-    ap_log_rerror(APLOG_MARK, APLOG_INFO, 0, r, "Failed to create enforcer");
-    ap_log_rerror(APLOG_MARK, APLOG_INFO, 0, r, "%s", err_msg);
-    return AUTHZ_DENIED;
+  if ( audience[0] ) {
+    aud_list[0] = audience;
+  } else  {
+    // Get the hostname for the audience. It is using hostname(NOT domain name). Set payload accordingly
+    if (gethostname(hostname, 1024) != 0) {
+      ap_log_rerror(APLOG_MARK, APLOG_INFO, 0, r, "Failed to get hostname");
+      return AUTHZ_DENIED;
+    }
+    aud_list[0] = hostname;
   }
```

Comment:
@djw8605 I'm talking about this block of code that prevents accepting the hostname as an audience if the "audience" option is set.

Reply:
Ah, so you want the option to have the hostname and the custom hostname.

What about an option to have multiple, comma separated audiences.  If it's not set, it's the hostname, if it is set, then it is only what is set (which can be the hostname?)

Reply:
That would work.

Reply:
@marcmengel Please go ahead and make that change to this PR before it gets merged.